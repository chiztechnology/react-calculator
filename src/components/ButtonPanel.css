.component-button-panel {
  background-color: #aeaeb4;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  flex: 1 0 auto;
}

.component-button-panel>div {
  width: 100%;
  margin-bottom: 1px;
  flex: 1 0 auto;
  display: flex;
}
