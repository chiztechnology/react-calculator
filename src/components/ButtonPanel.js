import React from 'react';
import Button from './Button'

class ButtonPanel extends React.Component {
  render() {
    return (
      <div className='component-button-panel'>
        <div>
          <Button name='AC' />
          <Button name='+/-' />
          <Button name='%' />
          <Button name='รท' orange />
        </div>
        <div>
          <Button name='7' />
          <Button name='8' />
          <Button name='9' />
          <Button name='x' orange />
        </div>
        <div>
          <Button name='4' />
          <Button name='5' />
          <Button name='6' />
          <Button name='-' orange />
        </div>
        <div>
          <Button name='1' />
          <Button name='2' />
          <Button name='3' />
          <Button name='+' orange />
        </div>
        <div>
          <Button name='0' wide />
          <Button name='.' />
          <Button name='=' orange />
        </div>
      </div>
    );
  }
}

export default ButtonPanel;