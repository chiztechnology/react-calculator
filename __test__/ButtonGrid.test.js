import React from 'react';
import renderer from 'react-test-renderer';

import ButtonPanel from '../src/components/ButtonPanel';

it('renders a Button correctly when there are no items', () => {
  const tree = renderer.create(<ButtonPanel clickHandler={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});
