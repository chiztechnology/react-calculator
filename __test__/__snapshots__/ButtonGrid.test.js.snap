// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Button correctly when there are no items 1`] = `
<div
  className="component-button-panel"
>
  <div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        AC
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        +/-
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        %
      </button>
    </div>
    <div
      className="component-button orange"
    >
      <button
        onClick={[Function]}
      >
        รท
      </button>
    </div>
  </div>
  <div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        7
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        8
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        9
      </button>
    </div>
    <div
      className="component-button orange"
    >
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        4
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        5
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        6
      </button>
    </div>
    <div
      className="component-button orange"
    >
      <button
        onClick={[Function]}
      >
        -
      </button>
    </div>
  </div>
  <div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        1
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        2
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        3
      </button>
    </div>
    <div
      className="component-button orange"
    >
      <button
        onClick={[Function]}
      >
        +
      </button>
    </div>
  </div>
  <div>
    <div
      className="component-button  wide"
    >
      <button
        onClick={[Function]}
      >
        0
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        .
      </button>
    </div>
    <div
      className="component-button orange"
    >
      <button
        onClick={[Function]}
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly when there are no items 1`] = `
<div
  className="component-button-panel"
>
  <div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        AC
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        +/-
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        %
      </button>
    </div>
    <div
      className="component-button orange"
    >
      <button
        onClick={[Function]}
      >
        รท
      </button>
    </div>
  </div>
  <div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        7
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        8
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        9
      </button>
    </div>
    <div
      className="component-button orange"
    >
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        4
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        5
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        6
      </button>
    </div>
    <div
      className="component-button orange"
    >
      <button
        onClick={[Function]}
      >
        -
      </button>
    </div>
  </div>
  <div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        1
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        2
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        3
      </button>
    </div>
    <div
      className="component-button orange"
    >
      <button
        onClick={[Function]}
      >
        +
      </button>
    </div>
  </div>
  <div>
    <div
      className="component-button  wide"
    >
      <button
        onClick={[Function]}
      >
        0
      </button>
    </div>
    <div
      className="component-button"
    >
      <button
        onClick={[Function]}
      >
        .
      </button>
    </div>
    <div
      className="component-button orange"
    >
      <button
        onClick={[Function]}
      >
        =
      </button>
    </div>
  </div>
</div>
`;
